### DISCUSSION TITLE
Quantify gene expression with kallisto

### DISCUSSION THREAD

**Discussion by nazerafauzia** [GuyTeichman/RNAlysis#6] [Q&A] [answered]
Hi Team,

I want to quantify the gene expression, but my experiment requires a different transcriptome that is not available in kallisto. In the A-to-Z tutorial Analysis #2 mentioned that I can index any FASTA file through RNAlysis, by entering the “FASTQ” menu -> “RNA sequencing quantification” -> “Create kallisto index…”. But I don't understand what kind of file I have to load. Is it the FASTA file of the full genome sequencing?

Thanks.
Nazera

**Comment by GuyTeichman** [comment, marked as answer]
Hi Nazera,
The file you need to load in order to create a _kallisto_ index is a FASTA of your organism's transcriptome. 
Meaning, each sequence in the FASTA file is a transcript. If a gene has multiple transcripts (e.g. alternative splicing), then each of those transcripts should be its own sequence in the FASTA file. 

Other aligners (such as HISAT2) can work with a reference genome instead of a reference transcriptome, but _kallisto_ was created to work specifically with transcriptomes. 

If you have any other questions, feel free to reply or open another thread. 

Good luck,
Guy.

**Comment by nazerafauzia** [comment]
Hi Guy,
Thank you for your reply.
I've tried to load my FASTA file (.fa), but it mentioned that _RNAlysis_ encountered the following error, which I don't understand.
Do you know how to solve this issue?

Thanks.
Nazera

Traceback (most recent call last):

  File "C:\Users\nazer\AppData\Local\Programs\Python\Python310\lib\site-packages\rnalysis\gui\gui.py", line 3425, in finish_generic_job
    raise worker_output[0]

  File "C:\Users\nazer\AppData\Local\Programs\Python\Python310\lib\site-packages\rnalysis\gui\gui_widgets.py", line 268, in run
    result = self.partial()

  File "C:\Users\nazer\AppData\Local\Programs\Python\Python310\lib\site-packages\rnalysis\fastq.py", line 70, in kallisto_create_index
    io.run_subprocess(call)

  File "C:\Users\nazer\AppData\Local\Programs\Python\Python310\lib\site-packages\rnalysis\utils\io.py", line 1294, in run_subprocess
    with subprocess.Popen(args, stdout=stdout, stderr=stderr) as process:

  File "C:\Users\nazer\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,

  File "C:\Users\nazer\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 1440, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,

FileNotFoundError: [WinError 2] The system cannot find the file specified

**Reply by GuyTeichman** [reply]
Hi Nazera,
My guess is that the path you specified to your kallisto installation directory is incorrect. Could you please send me a screen shot of your "create kallisto index" window just before you click "start"?

Guy

**Reply by nazerafauzia** [reply]
Hi Guy,
Here is the screenshot
![image](https://user-images.githubusercontent.com/121092236/209496053-c6c24fa8-4778-46e8-8793-ca873158b3e1.png)

Nazera

**Reply by GuyTeichman** [reply]
Hi,
If I'm not mistaken, the actual installation folder should be a folder named 'kallisto'. It is probably inside the directory you extracted your kallisto files into (meaning, inside the directory 'kallisto_windows-v0.46.1'). 
So, you should be pointing _RNAlysis_ to ".../kallisto_windows-v0.46.1/kallisto". 
Please try that and let me know if it works. 

Best,
Guy.

**Reply by nazerafauzia** [reply]
Hi,
It works.
Thank you so much!

Nazera
